src/Features/Core/Portable/UnusedReferences/UnusedReferencesRemover.cs

```diff
@@ -32,22 +34,24 @@ internal static class UnusedReferencesRemover
             CancellationToken cancellationToken)
         {
             var projects = solution.Projects
-                .Where(project => projectFilePath.Equals(project.FilePath, System.StringComparison.OrdinalIgnoreCase));
+                .Where(project => projectFilePath.Equals(project.FilePath, StringComparison.OrdinalIgnoreCase));
 
             HashSet<string> usedAssemblyFilePaths = new();
             HashSet<string> usedProjectFileNames = new();
 
-            foreach (var project in projects)
+            var getCompilationTasks = projects.Select(project => project.GetCompilationAsync(cancellationToken));
+            var compilations = await Task.WhenAll(getCompilationTasks).ConfigureAwait(false);
+
+            foreach (var compilation in compilations)
```

Comment:
it feels odd taht we, in parallel, ensure we're using a ton of memory.  but we then process that linearly.  Does processing one project depend on another?  if not, i would actually just process every project in parallel.